core/templates/dev/head/domain/exploration/SolutionObjectFactory.js

```diff
@@ -0,0 +1,125 @@
+      return new Solution(
+        solutionBackendDict.answer_is_exclusive,
+        solutionBackendDict.correct_answer,
+        solutionBackendDict.explanation);
+    };
+
+    Solution.createNew = function(
+      answerIsExclusive, correctAnswer, explanation) {
+      return new Solution(answerIsExclusive, correctAnswer, explanation);
+    };
+
+    Solution.prototype.getSummary = function(interactionId, choices) {
+      var isExclusiveAnswer = (
+        this.answerIsExclusive ? 'The only' : 'One');
+      var correctAnswer = '';
+      if (interactionId === 'GraphInput') {
+        correctAnswer = '[Graph Object]';
+      } else if (interactionId === 'MultipleChoiceInput') {
+        correctAnswer = (
+          oppiaHtmlEscaper.objToEscapedJson(
+            choices[this.correctAnswer].label));
+      } else if (interactionId === 'MathExpressionInput') {
+        correctAnswer = this.correctAnswer.latex;
+      } else if (interactionId === 'CodeRepl' ||
+        interactionId === 'PencilCodeEditor') {
+        correctAnswer = this.correctAnswer.code;
+      } else if (interactionId === 'MusicNotesInput') {
+        correctAnswer = '[Music Notes Object]';
+      } else if (interactionId === 'ImageClickInput') {
+        correctAnswer = this.correctAnswer.clickedRegions;
+      } else if (interactionId === 'LogicProof') {
+        correctAnswer = this.correctAnswer.correct;
+      } else {
+        correctAnswer = (
+          oppiaHtmlEscaper.objToEscapedJson(this.correctAnswer));
+      }
+      var explanation = (
+        $filter('convertToPlainText')(this.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    Solution.prototype.setCorrectAnswer = function (correctAnswer) {
+      this.correctAnswer = correctAnswer;
+    };
+
+    Solution.prototype.getCorrectAnswerHtml = function (objectType) {
+      if (objectType === 'CodeString') {
+        return this.correctAnswer.code;
+      } else if (objectType === 'ImageWithRegions') {
+        return this.correctAnswer.clickedRegions;
+      } else if (objectType === 'Graph') {
+        return '[Graph Object]';
+      } else if (objectType === 'UnicodeString') {
+        return this.correctAnswer.latex;
+      } else if (objectType === 'LogicQuestion') {
+        return this.correctAnswer.correct;
+      } else if (objectType === 'MusicPhrase') {
+        return '[Music Phrase Object]';
+      } else {
+        return this.correctAnswer;
+      }
+    };
+
+    Solution.prototype.getObjectEditorHtml = function(objectType) {
+      var element = $('<object-editor>');
+      element.attr('obj-type', objectType);
+      element.attr('init-args', '{choices: ruleDescriptionChoices}');
+      element.attr('is-editable', 'true');
+      element.attr('always-editable', 'true');
+      element.attr('style', 'color: black;');
+
+      if (objectType === 'UnicodeString') {
```

Comment:
Yeah, this is all quite confusing. If we can get the direct preview of the interaction HTML to work then hopefully we can remove a lot of this stuff.